tests/test_h1_server.c

```diff
@@ -0,0 +1,409 @@
+
+    test->request_num++;
+    aws_http_stream_configure_server_request_handler(stream, &options);
+    // TODO
+}
+
+static int s_tester_init(struct aws_allocator *alloc) {
+    aws_load_error_strings();
+    aws_common_load_log_subject_strings();
+    aws_io_load_error_strings();
+    aws_io_load_log_subject_strings();
+    aws_http_library_init(alloc);
+
+    AWS_ZERO_STRUCT(s_tester);
+
+    s_tester.alloc = alloc;
+
+    s_tester.request_num = 0;
+
+    struct aws_logger_standard_options logger_options = {
+        .level = AWS_LOG_LEVEL_TRACE,
+        .file = stderr,
+    };
+    ASSERT_SUCCESS(aws_logger_init_standard(&s_tester.logger, s_tester.alloc, &logger_options));
+    aws_logger_set(&s_tester.logger);
+
+    ASSERT_SUCCESS(testing_channel_init(&s_tester.testing_channel, alloc));
+
+    s_tester.server_connection = aws_http_connection_new_http1_1_server(alloc, SIZE_MAX);
+    ASSERT_NOT_NULL(s_tester.server_connection);
+    struct aws_http_server_connection_options options = AWS_HTTP_SERVER_CONNECTION_OPTIONS_INIT;
+    options.connection_user_data = &s_tester;
+    options.on_incoming_request = s_tester_on_incoming_request;
+
+    ASSERT_SUCCESS(aws_http_connection_configure_server(s_tester.server_connection, &options));
+
+    struct aws_channel_slot *slot = aws_channel_slot_new(s_tester.testing_channel.channel);
+    ASSERT_NOT_NULL(slot);
+    s_tester.server_connection->channel_slot = slot;
+    ASSERT_SUCCESS(aws_channel_slot_insert_end(s_tester.testing_channel.channel, slot));
+    ASSERT_SUCCESS(aws_channel_slot_set_handler(slot, &s_tester.server_connection->channel_handler));
+
+    aws_channel_acquire_hold(s_tester.testing_channel.channel);
+
+    testing_channel_drain_queued_tasks(&s_tester.testing_channel);
+
+    return AWS_OP_SUCCESS;
+}
+
+static int s_server_request_clean_up(void) {
+    for (int i = 0; i < s_tester.request_num; i++) {
+        aws_http_stream_release(s_tester.requests[i].request_handler);
+        aws_byte_buf_clean_up(&s_tester.requests[i].storage);
+    }
+    return AWS_OP_SUCCESS;
+}
+
+static int s_server_tester_clean_up(void) {
+    s_server_request_clean_up();
+    aws_http_connection_release(s_tester.server_connection);
+    ASSERT_SUCCESS(testing_channel_clean_up(&s_tester.testing_channel));
+    aws_http_library_clean_up();
+    aws_logger_clean_up(&s_tester.logger);
+
+    return AWS_OP_SUCCESS;
+}
+
+/* For sending an aws_io_message into the channel, in the write or read direction */
+static int s_send_message_ex(struct aws_byte_cursor data) {
+
+    struct aws_io_message *msg = aws_channel_acquire_message_from_pool(
+        s_tester.testing_channel.channel, AWS_IO_MESSAGE_APPLICATION_DATA, data.len);
+    ASSERT_NOT_NULL(msg);
+
+    ASSERT_TRUE(aws_byte_buf_write_from_whole_cursor(&msg->message_data, data));
+
+    ASSERT_SUCCESS(testing_channel_push_read_message(&s_tester.testing_channel, msg));
+
+    return AWS_OP_SUCCESS;
+}
+
+static int s_send_request_str(const char *str) {
```

Comment:
How about `s_send_message_c_str`? `send_request` implies it does something different than `send_message`.